lib/x509.mli

```diff
@@ -1017,6 +1017,18 @@ module Authenticator : sig
       over certificate pinning. *)
   val server_cert_fingerprint : time:(unit -> Ptime.t option) ->
     hash:Mirage_crypto.Hash.hash -> fingerprint:Cstruct.t -> t
+
+  val of_string : time:(unit -> Ptime.t option) -> string
+    -> (Authenticator.t, [> `Msg of string ]) result
```

Comment:
Would the following signature be sensible for your use cases:
```OCaml
  val of_string : string -> (unit -> Ptime.t option -> Authenticator.t, [> `Msg of string ]) result
```

The advantage would be that `Authenticator.of_string` could be used directly in cmdliner. The retrieval of the current timestamp would need to be provided only later.